### PR TITLE
refactor: remove low S check from util-crypto

### PR DESCRIPTION
**Pull request by jjyr**
Cause we already use witness, the low S rule for signature is unnecessary.

See https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki#new-rules

### CODE REVIEW

File: util/crypto/src/secp/signature.rs [hunk 1]
```diff
@@ -50,15 +50,6 @@ impl Signature {
         Signature(sig)
     }
 
-    /// Check if this is a "low" signature.
-    pub fn is_low_s(&self) -> bool {
```
Comment (janx) [inline util/crypto/src/secp/signature.rs]: where is this function used?

File: util/crypto/src/secp/signature.rs [hunk 2]
```diff
@@ -74,7 +65,7 @@ impl Signature {
                 return false;
             }
         };
-        self.v() <= 1 && h_r < N && h_r >= ONE && h_s < N && h_s >= ONE
+        self.v() < 4 && h_r < N && h_r >= ONE && h_s < N && h_s >= ONE
```
Comment (janx) [inline util/crypto/src/secp/signature.rs]: why is this change needed?

File: util/crypto/src/secp/signature.rs (hunk 1 above)
Comment (jjyr) [reply util/crypto/src/secp/signature.rs]: Not used in current CKB, this PR intends to remove it to avoid confusion.

File: util/crypto/src/secp/signature.rs (hunk 2 above)
Comment (jjyr) [reply util/crypto/src/secp/signature.rs]: After checking the source code of secp256k1 and peter wullies's answer, it seems in practice the `v` is only considered either `0` or `1`.

I'll rollback this.

https://github.com/bitcoin-core/secp256k1/blob/544435fc90a5672d862e2a51f44c10251893b97d/src/ecdsa_impl.h#L298

https://bitcoin.stackexchange.com/questions/38351/ecdsa-v-r-s-what-is-v

Comment (jjyr) [reply util/crypto/src/secp/signature.rs]: After checking the source code of secp256k1 and peter wullies's answer, it seems in practice the `v` is only considered either `0` or `1`.

I'll rollback this.

https://github.com/bitcoin-core/secp256k1/blob/544435fc90a5672d862e2a51f44c10251893b97d/src/ecdsa_impl.h#L298

https://bitcoin.stackexchange.com/questions/38351/ecdsa-v-r-s-what-is-v
